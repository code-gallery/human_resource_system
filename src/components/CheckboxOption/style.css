.CheckboxOption {
  box-sizing: border-box;
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
}

.CheckboxOption__input {
  margin-right: 5px;
}

.CheckboxOption__text {
  margin: 0;
  font-family: 'Graphik-Regular', sans-serif;
  -webkit-font-smoothing: antialiased;
  /* font-size: 10px; */
  line-height: 14px;
  margin-right: 10px;
  color: #7F8DAA;
}
