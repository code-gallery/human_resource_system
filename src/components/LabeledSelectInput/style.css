.LabeledSelectInput {
  box-sizing: border-box;
  position: relative;
  display: block;
  width: 100%;
  -webkit-font-smoothing: antialiased;
}

.LabeledSelectInput__txt {
  display: block;
  margin-bottom: 6.35px;
  font-size: 14px;
  line-height: 17px;
  font-family: "Graphik-Regular", sans-serif;
  color: #44484C;
}

.LabeledSelectInput__icon {
  position: absolute;
  bottom: 20px;
  right: 18px;
  z-index: 10;
  display: block;
  padding: 0;
  width: 9px;
}

.LabeledSelectInput__select {
  all: initial;
  box-sizing: border-box;
  position: relative;
  z-index: 10;
  display: block;
  width: 100%;
  padding: 13.45px 15px 10.25px 15px;
  border: 1px solid #E7E5EE;
  border-radius: 3px;
  font-size: 17px;
  line-height: 24px;
  font-family: "Graphik-Regular", sans-serif;
  -webkit-font-smoothing: antialiased;
  cursor: pointer;
  color: #212224;
  overflow: hidden;
  white-space: pre;
  text-overflow: ellipsis;
  -webkit-appearance: none;					   
}

.LabeledSelectInput__select:focus {
  border-color: #7F8DAA;
}
