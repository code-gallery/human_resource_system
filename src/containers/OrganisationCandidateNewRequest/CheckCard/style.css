.CheckCard {
  box-sizing: border-box;
  position: relative;
  padding: 22px 29px 29px 29px;
  -webkit-font-smoothing: antialiased;
  border: 1px solid #E7E5EE;
  border-radius: 2px;
  background-color: #fff;
}

.CheckCard__basics {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.CheckCard__header {
  margin: 0;
  margin-bottom: 6.85px;
  font-family: "Graphik-Medium", sans-serif;
  font-weight: 500;
  font-size: 18px;
  line-height: 22px;
  color: #212224;
}

.CheckCard__sub-heading {
  margin: 0;
  font-family: "Graphik-Regular", sans-serif;
  font-weight: 400;
  font-size: 13px;
  line-height: 24px;
  user-select: none;
  color: #7F8DAA;
}

.CheckCard__right-area {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.CheckCard__btn {
  width: 80px;
  margin-left: 1rem;
}

.CheckCard__price {
  font-size: 13px;
  line-height: 16px;
  font-family: "Graphik-Medium", sans-serif;
  font-weight: 600;
  letter-spacing: 0.055em;
  color: #7F8DAA;
}

.CheckCard__radios,
.CheckCard__options {
  margin-top: 10px;
  display: flex;
  flex-wrap: wrap;
  justify-content: left;
}

.CheckCard__selects {
  margin-top: 10px;
  /*display: flex; */
  flex-wrap: wrap;
  justify-content: left;
}

.CheckCard__opt {
  margin-bottom:5px;
}

.CheckCard__options1{
  margin-top: 10px;
}

.CheckCard__width {
  width: 500px;
}

.CheckCard__select {
  max-width: 300px;
  margin-bottom: 20px;
}

.CheckCard__select .LabeledSelectInput__txt {
  color: #7F8DAA;
  font-size: 15px;
  line-height: 24px;
}

/* .CheckCard__opt.is-last {
  margin-bottom: 0;
} */

.LabeledText__option{
  padding-right:  50px;
}
.LabeledText__div{
  display: inline-flex;
  margin-top: 10px;
}
