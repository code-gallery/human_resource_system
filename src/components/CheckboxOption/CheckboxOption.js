import React from 'react'
import PropTypes from 'prop-types'
import Checkbox from 'components/Checkbox'
import './style.css'

const CheckboxOption = ({ checked, text, onChange, className }) => (
  <div className={`CheckboxOption ${className}`}>
    <Checkbox
      className="CheckboxOption__input"
      checked={checked}
      onChange={onChange}
    />

    <p className="CheckboxOption__text">{text}</p>

  </div>

)

CheckboxOption.propTypes = {
  text: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
  className: PropTypes.string
}

CheckboxOption.defaultProps = {
  className: '',
  checked: false
}

export default CheckboxOption
