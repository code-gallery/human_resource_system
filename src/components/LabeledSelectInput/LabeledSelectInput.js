import React from 'react'
import PropTypes from 'prop-types'
import Icon from 'components/Icon'
import './style.css'

const LabeledSelectInput = ({ className, label, onValueChange, options, value, placeholder,required }) => {
  let selectElem

  const handleChange = () => {
    onValueChange(selectElem.value)
  }

  const handleInputElem = (elem) => {
    selectElem = elem
  }

  return (
    <label className={`LabeledSelectInput ${className}`}>
      <span className="LabeledSelectInput__txt">{label}</span>
      <Icon className="LabeledSelectInput__icon" type="caretDown" color="44484c" />

      <select
        className="LabeledSelectInput__select"
        onChange={handleChange}
        ref={handleInputElem}
        value={value}
        required={required}
      >
		 {placeholder !== undefined ?
        <option className="placeholder__Text" key={placeholder} value="" disabled>{placeholder}</option>
        :null}																																		  
        {
          Object.entries(options).map(([ value, name ]) => (
            <option key={value} value={value}>{name}</option>
          ))
        }
      </select>
    </label>
  )
}

LabeledSelectInput.propTypes = {
  label: PropTypes.string.isRequired,
  onValueChange: PropTypes.func.isRequired,
  className: PropTypes.string,
  options: PropTypes.object.isRequired,
  value: PropTypes.string.isRequired
}

LabeledSelectInput.defaultProps = {
  className: '',
  required:false
}

export default LabeledSelectInput
