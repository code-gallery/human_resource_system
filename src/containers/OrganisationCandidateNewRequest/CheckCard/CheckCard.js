import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import OutlineButton from 'components/OutlineButton'
import CheckboxOption from 'components/CheckboxOption'
import LabeledSelectInput from 'components/LabeledSelectInput'
import prettyPrice from 'utils/prettyPrice'
import LabeledTextInput from 'components/LabeledTextInput'
import './style.css'


/**
 @NOTE: Currently this presentational component is being used to create both
 simple request check cards and ones with options (i.e Enchanced DBS checks).
 */
function checkvalue(onPersonaChange){
  onPersonaChange()

}
const CheckCard = ({
  requestName,
  verifier,
  price,
  added,
  options,
  selects,
  onChange,
  className,
  personaid,
  onPersonaChange
}) => (
  <article className={`CheckCard ${className}`}>
    <div  className="CheckCard__basics">
      <div className= "CheckCard__width">
        <h1 className="CheckCard__header">{requestName}</h1>
        <p className="CheckCard__sub-heading">Verified by {verifier}</p>
      </div>





     { console.log("personaid",personaid)}

      <div className="CheckCard__right-area">

      { options && options.length > 0 && (requestName ==='Biometric Identity Checks' || requestName.includes("DBS") ||requestName.includes("Scotland") ) &&
      <div className="CheckCard__options1">
        {options.map((option, idx) => {
          const isLast = idx === options.length - 1
          const style = classNames(
            'CheckCard__opt',
            { 'is-last': isLast }
          )
          {console.log("option", option)}
        {if(option.text == 'userPays') {
          var textname = 'User Pays'
          return (
            <CheckboxOption
              className={style}
              key={idx}
              text={textname}
              checked={option.checked}
              onChange={option.onValueChange}
            />
          ) }}
        }) }
      </div>
    }

        <span className="CheckCard__price">{prettyPrice(price)}</span>
        {!personaid || personaid==='No Persona' ?<div className="CheckCard__btn" >
         {

            added
              ? <OutlineButton color="red" onClick={onChange}>REMOVE</OutlineButton>
              : <OutlineButton color="blue" onClick={onChange}>ADD</OutlineButton>
          }


        </div>: checkvalue(onPersonaChange)}
      </div>
    </div>

    { selects   && selects.length > 0 && requestName !== 'Company & VAT Checks' &&
      <div className="CheckCard__selects">
        {selects.map((select) => (
          <LabeledSelectInput
            className={'CheckCard__select'}
            key={select.label}
            label={select.label}
            onValueChange={select.onValueChange}
            options={select.options}
            value={select.value}
          />
        ))}
      </div>
    }




      { options && options.length > 0 && requestName ==='Standard DBS Check' &&
      <div className="CheckCard__options1">
        {options.map((option, idx) => {
          const isLast = idx === options.length - 1
          const style = classNames(
            'CheckCard__opt',
            { 'is-last': isLast }
          )
          {if(option.text != 'userPays') {
          return (
            <CheckboxOption
              className={style}
              key={idx}
              text={option.text}
              checked={option.checked}
              onChange={option.onValueChange}
            />
          )}}
        }) }
      </div>
    }





    { options && options.length > 0 && requestName !=='Standard DBS Check'  &&
      <div  className={`box ${requestName==='Enhanced DBS Check' ? "CheckCard__options1" : "CheckCard__options"}`}>
        {options.map((option, idx) => {
          const isLast = idx === options.length - 1
          //console.log("option and idx" , option , idx)
          //console.log("this.state " , this.state)

          const style = classNames(
            'CheckCard__opt',
            'CheckCard__options_dbs',
            { 'is-last': isLast }
          )
          {if(option.text != 'userPays') {
          return (
            <CheckboxOption
              className={style}
              key={idx}
              text={option.text}
              checked={option.checked}
              onChange={option.onValueChange}

            />

           )}}
        }) }

      </div>
    }

{ selects   && selects.length > 0 && requestName === 'Company & VAT Checks' &&
      <div className="LabeledText__div" >
        {selects.map((select) => (
        <div className='LabeledText__option'><LabeledTextInput
        className="CandidateNewRequests__input"
        label={select.label}
        labelCurr={select.labelCurr}
        //placeholder={select.value}
        fontBold = {true}
        onValueChange={select.onValueChange}
        //onValueChange={val =>  this.setState({ role: val }, () => {this.getChecks1()})}
       // key = {this.state.resetRole}
      /></div>
        ))}
      </div>
    }



  </article>
)

CheckCard.propTypes = {
  requestName: PropTypes.string.isRequired,
  verifier: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  added: PropTypes.bool.isRequired,
  options: PropTypes.array,
  selects: PropTypes.array,
  onChange: PropTypes.func.isRequired,
  className: PropTypes.string
}

CheckCard.defaultProps = {
  className: '',
  options: null,
  selects: null
}

export default CheckCard
