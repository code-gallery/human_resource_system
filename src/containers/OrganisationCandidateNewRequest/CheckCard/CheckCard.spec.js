import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import CheckCard from './CheckCard'
import CheckboxOption from 'components/CheckboxOption'
import LabeledSelectInput from 'components/LabeledSelectInput'

describe('<CheckCard />', () => {
  const props = {
    requestName: 'DBS',
    verifier: 'Apii',
    price: 2500,
    added: false,
    onChange: jest.fn()
  }

  describe('Rendering', () => {
    it('renders', () => {
      const tree = renderer.create(<CheckCard {...props} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('When options are used', () => {
    const optionProps = {
      ...props,
      options: [
        {
          text: 'Test',
          checked: true,
          onValueChange: jest.fn()
        }
      ]
    }

    it('renders', () => {
      const tree = renderer.create(<CheckCard {...optionProps} />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('includes a checkbox with the correct text', () => {
      const wrapper = shallow(<CheckCard {...optionProps} />)
      const checkbox = wrapper.find(CheckboxOption)
      expect(checkbox.length).toEqual(1)
      expect(checkbox.prop('text')).toEqual('Test')
    })
  })

  describe('When selects are used', () => {
    const selectProps = {
      ...props,
      selects: [
        {
          label: 'Test',
          onValueChange: jest.fn(),
          value: 'val',
          options: {
            val: 'Value'
          }
        }
      ]
    }

    it('renders', () => {
      const tree = renderer.create(<CheckCard {...selectProps} />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('includes a select with the correct options', () => {
      const wrapper = shallow(<CheckCard {...selectProps} />)
      const select = wrapper.find(LabeledSelectInput)
      expect(select.length).toEqual(1)
      expect(select.prop('label')).toEqual('Test')
      expect(select.prop('options')).toEqual({ val: 'Value' })
    })
  })
})
