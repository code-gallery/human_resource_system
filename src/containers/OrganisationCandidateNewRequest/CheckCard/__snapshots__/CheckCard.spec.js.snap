// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckCard /> Rendering renders 1`] = `
<article
  className="CheckCard "
>
  <div
    className="CheckCard__basics"
  >
    <div>
      <h1
        className="CheckCard__header"
      >
        DBS
      </h1>
      <p
        className="CheckCard__sub-heading"
      >
        Verified by 
        Apii
      </p>
    </div>
    <div
      className="CheckCard__right-area"
    >
      <span
        className="CheckCard__price"
      >
        £25.00
      </span>
      <div
        className="CheckCard__btn"
      >
        <button
          className="OutlineButton is-blue "
          onClick={[Function]}
          type="button"
        >
          ADD
        </button>
      </div>
    </div>
  </div>
</article>
`;

exports[`<CheckCard /> When options are used renders 1`] = `
<article
  className="CheckCard "
>
  <div
    className="CheckCard__basics"
  >
    <div>
      <h1
        className="CheckCard__header"
      >
        DBS
      </h1>
      <p
        className="CheckCard__sub-heading"
      >
        Verified by 
        Apii
      </p>
    </div>
    <div
      className="CheckCard__right-area"
    >
      <span
        className="CheckCard__price"
      >
        £25.00
      </span>
      <div
        className="CheckCard__btn"
      >
        <button
          className="OutlineButton is-blue "
          onClick={[Function]}
          type="button"
        >
          ADD
        </button>
      </div>
    </div>
  </div>
  <div
    className="CheckCard__options"
  >
    <div
      className="CheckboxOption CheckCard__opt is-last"
    >
      <div
        className="Checkbox CheckboxOption__input"
      >
        <label
          className="Checkbox__label is-checked"
        >
          <input
            checked={true}
            className="Checkbox__input"
            onChange={[Function]}
            type="checkbox"
          />
        </label>
      </div>
      <p
        className="CheckboxOption__text"
      >
        Test
      </p>
    </div>
  </div>
</article>
`;

exports[`<CheckCard /> When selects are used renders 1`] = `
<article
  className="CheckCard "
>
  <div
    className="CheckCard__basics"
  >
    <div>
      <h1
        className="CheckCard__header"
      >
        DBS
      </h1>
      <p
        className="CheckCard__sub-heading"
      >
        Verified by 
        Apii
      </p>
    </div>
    <div
      className="CheckCard__right-area"
    >
      <span
        className="CheckCard__price"
      >
        £25.00
      </span>
      <div
        className="CheckCard__btn"
      >
        <button
          className="OutlineButton is-blue "
          onClick={[Function]}
          type="button"
        >
          ADD
        </button>
      </div>
    </div>
  </div>
  <div
    className="CheckCard__selects"
  >
    <label
      className="LabeledSelectInput CheckCard__select"
    >
      <span
        className="LabeledSelectInput__txt"
      >
        Test
      </span>
      <svg
        className="LabeledSelectInput__icon"
        viewBox="0 0 8 8"
      >
        <path
          d="M0 2l4 4 4-4h-8z"
          fill="44484c"
        />
      </svg>
      <select
        className="LabeledSelectInput__select"
        onChange={[Function]}
        value="val"
      >
        <option
          value="val"
        >
          Value
        </option>
      </select>
    </label>
  </div>
</article>
`;
